de.metas.vertical.creditscore/creditpass/src/main/java/de/metas/vertical/creditscore/creditpass/repository/CreditPassConfigRepository.java

```diff
@@ -0,0 +1,154 @@
+ * published by the Free Software Foundation, either version 2 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public
+ * License along with this program. If not, see
+ * <http://www.gnu.org/licenses/gpl-2.0.html>.
+ * #L%
+ */
+
+import de.metas.bpartner.BPartnerId;
+import de.metas.bpartner.service.IBPGroupDAO;
+import de.metas.cache.CCache;
+import de.metas.money.CurrencyId;
+import de.metas.util.Services;
+import de.metas.vertical.creditscore.base.model.I_CS_Transaction_Result;
+import de.metas.vertical.creditscore.base.spi.model.ResultCode;
+import de.metas.vertical.creditscore.creditpass.CreditPassConstants;
+import de.metas.vertical.creditscore.creditpass.model.*;
+import de.metas.vertical.creditscore.creditpass.model.extended.I_C_Order;
+import org.adempiere.ad.dao.IQueryBL;
+import org.adempiere.service.ISysConfigBL;
+import org.adempiere.user.UserId;
+import org.compiere.model.I_C_BP_Group;
+import org.compiere.util.Env;
+import org.springframework.stereotype.Repository;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Repository
+public class CreditPassConfigRepository
+{
+
+	private final CCache<BPartnerId, CreditPassConfig> configCache = CCache.<BPartnerId, CreditPassConfig>builder()
+			.initialCapacity(1)
+			.tableName(I_CS_Creditpass_Config.Table_Name)
+			.additionalTableNameToResetFor(I_C_BP_Group.Table_Name)
+			.additionalTableNameToResetFor(I_CS_Creditpass_Config_PaymentRule.Table_Name)
+			.additionalTableNameToResetFor(I_CS_Creditpass_CP_Fallback.Table_Name)
+			.additionalTableNameToResetFor(I_CS_Transaction_Result.Table_Name)
+			.additionalTableNameToResetFor(I_C_Order.Table_Name)
+			.build();
+
+	public CreditPassConfig getConfigByBPartnerId(BPartnerId businessPartnerId)
+	{
+		return configCache.getOrLoad(businessPartnerId, () -> getByBPartnerId(businessPartnerId));
+	}
+
+	private CreditPassConfig getByBPartnerId(BPartnerId businessPartnerId)
+	{
+
+		final IBPGroupDAO bpGroupRepo = Services.get(IBPGroupDAO.class);
+		final I_C_BP_Group bpGroup = bpGroupRepo.getByBPartnerId(businessPartnerId);
+
+		final IQueryBL queryBL = Services.get(IQueryBL.class);
+
+		final I_CS_Creditpass_Config config = queryBL
+				.createQueryBuilder(I_CS_Creditpass_Config.class)
+				.addOnlyActiveRecordsFilter()
+				.orderBy(I_CS_Creditpass_Config.COLUMN_Created).create().list()
+				.stream().filter(c -> queryBL
+						.createQueryBuilder(I_CS_Creditpass_BP_Group.class)
+						.addOnlyActiveRecordsFilter()
+						.addEqualsFilter(I_CS_Creditpass_BP_Group.COLUMN_CS_Creditpass_Config_ID, c.getCS_Creditpass_Config_ID())
+						.create().list()
```

Comment:
note: `IQuery` has a `stream()` method with does `list().stream()` for you (but for me it's ok like this)